numpy/_core/src/multiarray/dtypemeta.c

```diff
@@ -1131,11 +1132,12 @@ dtypemeta_wrap_legacy_descriptor(_PyArray_LegacyDescr *descr,
         /* Further fields are not common between DTypes */
     };
     memcpy(dtype_class, &prototype, sizeof(PyArray_DTypeMeta));
-    /* Fix name of the Type*/
+    /* Fix name and superclass of the Type*/
     ((PyTypeObject *)dtype_class)->tp_name = name;
+    ((PyTypeObject *)dtype_class)->tp_base = dtype_super_class,
     dtype_class->dt_slots = dt_slots;
 
-    /* Let python finish the initialization (probably unnecessary) */
```

Comment:
It is actually necessary - without it, `numpy/_core/tests/test_dtype.py::TestUserDType::test_custom_structured_dtype` fails, in a way that is most puzzling: the `assert dt.type` fails with an `AtttributeError`, but if one tries debugging, the `assert` passes.

So, we better keep it!